app/models/template.rb

```diff
@@ -114,6 +114,8 @@ def self.parse_metadata(text)
   #   :force - set to true if you want to bypass locked templates
   #   :associate - either 'new', 'always' or 'never', determines when the template should associate objects based on metadata
   #   :lock - lock imported templates (false by default)
+  #   :lock_predicate - a predicate that evaluates to boolean, determines whether a template should be locked
+  #                   - use instead of :lock when more complex logic for locking is required
```

Comment:
Wouldn't it be more railsy to just use `lock`?


e.g. 
```ruby
options[:lock].respond_to?(:call) ? options[:lock].call(self) : !!options[:lock]
```

So you could pass:

```ruby
{lock: true}
```
or

```ruby
{lock: {|t| !template.new_record? && template.locked }
```